packages/editor-ui/src/composables/useRunWorkflow.ts

```diff
@@ -150,21 +150,24 @@ export function useRunWorkflow(useRunWorkflowOpts: { router: ReturnType<typeof u
 
 			let { runData: newRunData } = consolidatedData;
 			let executedNode: string | undefined;
+			let preferredTrigger: IStartRunData['preferredTrigger'];
 			if (
 				startNodeNames.length === 0 &&
 				'destinationNode' in options &&
 				options.destinationNode !== undefined
 			) {
 				executedNode = options.destinationNode;
 				startNodeNames.push(options.destinationNode);
-			} else if ('triggerNode' in options && 'nodeData' in options) {
+			} else if (options.triggerNode && options.nodeData) {
```

Comment:
There was no need to use `'...' in object` as a guard and it made the type casting necessary. Changing this get's rid of the type casting below.